aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1221,6 +1222,33 @@ static void addmm_impl_cpu_(
   }
 }
 
+// TODO: This should work through the structured kernel approach of TORCH_IMPL_FUNC
+// TODO: ensure devices are handled
+Tensor addmm_zerotensor(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha){
+  TORCH_INTERNAL_ASSERT(self._is_zerotensor() || mat1._is_zerotensor() || mat2._is_zerotensor() || beta.equal(0) || alpha.equal(0));
```

Comment:
Is this something we shouldn't need?

Reply:
this is not needed since `addmm_zerotensor` would only be called when one of the inputs is a ZT

Reply:
Ideally we'd like to call into this if the scalars are zero as well.